terraform/deployments/github/main.tf

```diff
@@ -115,6 +115,44 @@ resource "github_team_repository" "govuk_repos" {
   permission = "push"
 }
 
+resource "github_repository" "govuk_repos" {
+  for_each = { for repo in local.auto_configurable_repos : repo.name => repo }
+
+  name        = each.value.name
+  description = each.value.description
+
+  allow_squash_merge = true
+  allow_merge_commit = false
+  allow_rebase_merge = false
+
+  delete_branch_on_merge = true
+}
+
+resource "github_branch_protection" "govuk_repos" {
+  for_each = github_repository.govuk_repos
+
+  repository_id    = each.value.name
+  pattern          = "main"
+  enforce_admins   = true
+  allows_deletions = true
+
+  required_pull_request_reviews {
+    required_approving_review_count = 1
+  }
+
+  required_status_checks {
+    strict = true # This ensures the branch is up to date before merging
+    contexts = [
+      "Lint SCSS / Run Stylelint",
+      "Security Analysis / Run Brakeman",
+      "Lint Ruby / Run RuboCop",
+      "Lint JavaScript / Run Standardx",
+      "Test JavaScript / Run Jasmine",
+      "Test Ruby / Run RSpec"
```

Comment:
Unfortunately the checks are not the same for every repo, see: https://github.com/alphagov/govuk-saas-config/blob/main/github/repo_overrides.yml